.stylelintrc.json

```diff
@@ -0,0 +1,102 @@
+{
+  "extends": "stylelint-config-recommended",
+  "plugins": [
+    "stylelint-order"
+  ],
+  "ignoreFiles": [
+    "components/demo/code-dark-plus-styles.js",
+    "components/demo/code-tomorrow-styles.js",
+    "components/list/**/*.js"
+  ],
+  "rules": {
+    "at-rule-name-case": "lower",
+    "at-rule-name-space-after": "always-single-line",
+    "at-rule-semicolon-newline-after": "always",
+    "block-closing-brace-newline-after": "always",
+    "block-closing-brace-newline-before": "always-multi-line",
+    "block-closing-brace-space-before": "always-single-line",
+    "block-opening-brace-newline-after": "always-multi-line",
+    "block-opening-brace-space-after": "always-single-line",
+    "block-opening-brace-space-before": "always",
+    "color-hex-case": "lower",
+    "color-hex-length": "long",
+    "comment-whitespace-inside": "always",
+    "custom-property-pattern": [ "^d2l-", {
```

Comment:
This rule (regex) will allow selectors for custom elements such as `d2l-icon { ... }`.  Otherwise it will flag them as unknown.